agent/bench-scripts/postprocess/process-iteration-samples

```diff
@@ -220,24 +227,33 @@ foreach $sample (@json_samples) { # samples 0..N
 						if ($key eq get_label('value_label')) {
 							my %samples_hash;
 							$samples_hash{get_label('value_label')} = $sample_value;
-							if ($sample_hash{$metric_type}{$metric_name}[$i]{get_label('timeseries_label')}) {
+							if (exists($sample_hash{$metric_type}{$metric_name}[$i]{get_label('timeseries_label')})) {
 								$samples_hash{get_label('timeseries_label')} = $sample_hash{$metric_type}{$metric_name}[$i]{get_label('timeseries_label')};
 							}
-							if ( not defined $iteration{$metric_type}{$metric_name}[$iteration_matching_index]{get_label('samples_label')} ) {
+							if (exists($iteration{$metric_type}{$metric_name}[$iteration_matching_index]{get_label('samples_label')})) {
```

Comment:
I don't know if `exists(foo)` and `foo` are equivalent in an `if`, but I was assuming that they are and that you just added `exists` everywhere where that pattern held. But this change does not follow the pattern and seems broken.

Reply:
Good catch.  This is incorrect and simply adding a `not` to restore the proper test works to restore the unit tests to a passing state.